File: c/medium/find-unique-binary-string.c
#include <stdlib.h>

char* findDifferentBinaryString(char** binary_strings, int n) {
    char* result = (char*)malloc((n + 1) * sizeof(char));

    for (int i = 0; i < n; i++) {
        result[i] = (binary_strings[i][i] == '0') ? '1' : '0';
    }

    result[n] = '\0';
    return result;
}

File: rust/medium/find-unique-binary-string.rs

pub fn find_different_binary_string(binary_strings: Vec<String>) -> String {
    let n = binary_strings.len();
    let mut result = Vec::with_capacity(n);

    for (i, s) in binary_strings.iter().enumerate() {
        let bit = s.as_bytes()[i];
        result.push(if bit == b'0' { b'1' } else { b'0' });
    }

    String::from_utf8(result).unwrap()
}
